huaweicloud/services/rds/data_source_huaweicloud_rds_parametergroups.go

```diff
@@ -0,0 +1,182 @@
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+	return &sc
+}
+
+func DataSourceParametergroupsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	var (
+		listConfigurationsHttpUrl = "v3/{project_id}/configurations"
+		listConfigurationsProduct = "rds"
+	)
+	listConfigurationsClient, err := cfg.NewServiceClient(listConfigurationsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating Configuration Client: %s", err)
+	}
+
+	listConfigurationsPath := listConfigurationsClient.Endpoint + listConfigurationsHttpUrl
+	listConfigurationsPath = strings.ReplaceAll(listConfigurationsPath, "{project_id}", listConfigurationsClient.ProjectID)
+	getConfigurationsOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	listConfigurationsResp, err := listConfigurationsClient.Request("GET",
+		listConfigurationsPath, &getConfigurationsOpt)
+	if err != nil {
+		return diag.Errorf("error retrieving RDS configurations")
+	}
+	listConfigurationsRespBody, err := utils.FlattenResponse(listConfigurationsResp)
+	if err != nil {
+		return diag.Errorf("error returns the api response body: %s", err)
+	}
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(uuid)
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("configurations", flattenConfigurationsBody(listConfigurationsRespBody, d)),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenConfigurationsBody(resp interface{}, d *schema.ResourceData) []interface{} {
+	if resp == nil {
+		return nil
+	}
+	curJson := utils.PathSearch("configurations", resp, make([]interface{}, 0))
+	if curJson == nil {
+		return nil
+	}
+	configurationsName, nameOk := d.GetOk("name")
+	configurationsDatastoreVersionName, datastoreVersionNameOk := d.GetOk("datastore_version_name")
+	configurationsDatastoreName, datastoreNameOk := d.GetOk("datastore_name")
+	configurationsUserDefined, userDefinedOk := d.GetOk("user_defined")
+	curArray := curJson.([]interface{})
+	rst := make([]interface{}, 0, len(curArray))
+	for _, v := range curArray {
+		name := utils.PathSearch("name", v, "")
+		datastoreVersionName := utils.PathSearch("datastore_version_name", v, "")
+		datastoreName := utils.PathSearch("datastore_name", v, "")
+		userDefined := utils.PathSearch("user_defined", v, "").(bool)
+		if nameOk && configurationsName.(string) != name.(string) {
+			continue
+		}
+		if datastoreVersionNameOk && configurationsDatastoreVersionName.(string) != datastoreVersionName.(string) {
+			continue
+		}
+		if datastoreNameOk && configurationsDatastoreName.(string) != datastoreName.(string) {
+			continue
+		}
+		if userDefinedOk && configurationsUserDefined.(bool) != userDefined {
```

Comment:
if userDefinedOk is false, then this judge is false forever

Reply:
done